helios-retrofit-integration/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt

```diff
@@ -0,0 +1,17 @@
+package helios.retrofit
+
+import arrow.core.flatMap
+import helios.core.Json
+import helios.typeclasses.Decoder
+import okhttp3.ResponseBody
+import retrofit2.Converter
+import java.nio.ByteBuffer
+import java.nio.CharBuffer
+
+class HeliosResponseBodyConverter<T>(private val decoder: Decoder<T>) : Converter<ResponseBody, T> {
+  override fun convert(value: ResponseBody): T? = Json
+    .parseFromByteBuffer(ByteBuffer.wrap(value.byteStream().readBytes()))
+    .flatMap { it.decode(decoder) }
+    .fold({ null }, { it })
+    .also { value.close() }
```

Comment:
Try/catch/finally for the stream?

Reply:
And what should I do in the catch? Returning null as well?

Reply:
exactly!